tooling/minimize/src/program.rs

```diff
@@ -64,6 +64,17 @@ impl<'tcx> Ctxt<'tcx> {
             globals: self.globals,
         }
     }
+
+    // Returns FnName associated with some key. If it does not exist it creates a new one.
+    pub fn get_fn_name_or_set(&mut self, key: (rs::DefId, &'tcx rs::List<rs::GenericArg<'tcx>>)) -> FnName {
+        if !self.fn_name_map.contains_key(&key) {
```

Comment:
This is a great usecase for `fn_name_map.entry(&key).or_insert_with` :)

Reply:
I use it now but we precompute the `length` since the borrow checker complains and the length is some what ambiguous while the `Entry` exists.

Reply:
Ah, that's unfortunate, so it's not as nice as I had hoped. Oh well.